graphdatascience/query_runner/gds_arrow_client.py

```diff
@@ -537,11 +553,28 @@ def upload_triplets(
         """
         self._upload_data(graph_name, "triplet", triplet_data, batch_size, progress_callback)
 
+    def __getstate__(self) -> Dict[str, Any]:
+        state = self.__dict__.copy()
+        # Remove the FlightClient as it isn't serializable
+        if "_flight_client" in state:
+            del state["_flight_client"]
+        return state
+
+    def _client(self) -> flight.FlightClient:
+        """
+        Lazy client construction to help pickle this class because a PyArrow
+        FlightClient is not serializable.
+        """
+        if not hasattr(self, "_flight_client") or not self._flight_client:
+            self._flight_client = self._instantiate_flight_client()
+        return self._flight_client
+
     def _send_action(self, action_type: str, meta_data: dict[str, Any]) -> dict[str, Any]:
         action_type = self._versioned_action_type(action_type)
 
         try:
-            result = self._flight_client.do_action(flight.Action(action_type, json.dumps(meta_data).encode("utf-8")))
+            client = self._client()
```

Comment:
so this either just gets the field, or if this class was restored from a serialised state, then it re-initialises the flight client. Cool.

Reply:
yep